applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -197,6 +229,12 @@ class CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4 : public Explic
         // Call the base RK4 finalize substep method
         BaseType::Initialize();
 
+        // Initialize the postprocess non-historical variables
+        for (auto &r_node : r_model_part.Nodes()) {
+            r_node.SetValue(MACH, 0.0);
+            r_node.SetValue(SOUND_VELOCITY, 0.0);
+        }
+
         // If required, initialize the OSS projection variables
         if (r_process_info[OSS_SWITCH]) {
```

Comment:
since the processinfo is not const this will allocate the `OSS_SWITCH` if it is not already in the ProcessInfo

Reply:
The process info must always contain this (It is set in the solver)

Reply:
This seems like quite an implementation detail that is dangerous and couples (too) tightly the python solver and the strategy

I recommend the following: since it is required that OSSSWITCH is in the ProcessInfo I would access it here as const and in the Check fct I would make sure that the ProcessInfo Has it

Reply:
Makes sense. Thanks for the hint 😉

Reply:
Check added (the `ProcessInfo` was already `const` in here.